nav2_recoveries/plugins/back_up.cpp

```diff
@@ -120,11 +120,12 @@ bool BackUp::isCollisionFree(
   double sim_position_change;
   const double diff_dist = abs(command_x_) - distance;
   const int max_cycle_count = static_cast<int>(cycle_frequency_ * simulate_ahead_time_);
-
+  geometry_msgs::msg::Pose2D init_pose = pose2d;
```

Comment:
Did you test this to fix your issue with too long looking back?

Reply:
Yes, manual test using tb3 simulation.